### ISSUE TITLE
Feature Request: Global learning rate schedule

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#670]
Idea: ability to set global learning rate according to a schedule. This would then be used in places like SGDUpdater, RMSProp etc.
Ideally, this would be implemented as an interface to support various types of learning rate schedules. For example,
- Step decay: reduce learning rate in steps as certain number of training iterations have completed
- Exponential decay (l0 * exp(-alpha * numIteratons))
- 1/t decay
- Delayed versions of the above: i.e., start reducing learning rate after 5 epochs etc.
- Custom schedules (user-supplied class implementing an interface)

**Comment by agibsonccc** [comment]
This was implemented.
